fhd_core/source_modeling/source_dft_multi.pro

```diff
@@ -50,78 +52,43 @@ FOR a_i=0L,n_alpha-1 DO BEGIN
 ENDFOR
 
 IF keyword_set(gaussian_source_models) then begin
-  if tag_exist(source_array, 'shape') then begin
-    gaussian_ra=make_array(n_elements(source_array),/double)
-    gaussian_dec=make_array(n_elements(source_array),/double)
-    gaussian_rot=make_array(n_elements(source_array),/double)
-    for source_ind=0,n_elements(source_array)-1 do begin
-      if tag_exist(source_array[source_ind], 'shape') then begin
+    if tag_exist(source_array, 'shape') then begin
+        gauss_test = ceil(0>(source_array.shape.x + source_array.shape.y)<1)
+        gauss_inds = where(gauss_test GT 0,n_gauss)
+    endif else n_gauss=0
+
+    if n_gauss GT 0 then begin
         ;Convert from FWHM in arcsec to stddev in deg
-        gaussian_ra[source_ind]=source_array[source_ind].shape.x/(7200*sqrt(2*alog(2)))
-        gaussian_dec[source_ind]=source_array[source_ind].shape.y/(7200*sqrt(2*alog(2)))
+        gaussian_ra=double(source_array[gauss_inds].shape.x)/(7200*sqrt(2*alog(2)))
+        gaussian_dec=double(source_array[gauss_inds].shape.y)/(7200*sqrt(2*alog(2)))
         ;Convert from deg to rad
-        gaussian_rot[source_ind]=source_array[source_ind].shape.angle*!Pi/180.
-      endif else begin
-        gaussian_ra[source_ind]=0
-        gaussian_dec[source_ind]=0
-        gaussian_rot[source_ind]=0
-      endelse
-    endfor
-    inds_gauss=where(gaussian_ra,n_gauss_params,complement=inds_point)
-    if n_gauss_params eq 0 then begin
-      print, 'Catalog does not contain Gaussian shape parameters. Unsetting keyword gaussian_source_models.'
-      undefine, gaussian_source_models
-    endif else begin
+        gaussian_rot=double(source_array[gauss_inds].shape.angle)*!DPi/180.
+
+        ;Compression due to flat sky approximation of curved sky, see J Cook et al. 2022
+        Eq2Hor,source_array[gauss_inds].ra,source_array[gauss_inds].dec, obs.JD0, gauss_alt, gauss_az, nutate=1,precess=1,$
+          aberration=0, refract=0, lon=obs.lon, alt=obs.alt, lat=obs.lat
+        gaussian_ra_corr = sqrt( sin(gaussian_rot)^2 + cos(gaussian_rot)^2 * sin((90.-gauss_az) * !DPi/180.)^2 )
+        gaussian_dec_corr = sqrt( cos(gaussian_rot)^2 + sin(gaussian_rot)^2 * cos((90.-gauss_az) * !DPi/180.)^2 )
+
+        ;Calculate the x,y pixel coordinates of the gaussian sources
+        apply_astrometry, obs, x_arr=gaussian_x_vals, y_arr=gaussian_y_vals, ra_arr=[[obs.obsra-0.5*gaussian_ra], $
+          [obs.obsra+0.5*gaussian_ra]], dec_arr=[[obs.obsdec-0.5*gaussian_dec],[obs.obsdec+0.5*gaussian_dec]], /ad2xy
+        gaussian_x = (gaussian_x_vals[*,1]-gaussian_x_vals[*,0]) / gaussian_ra_corr
+        gaussian_y = (gaussian_y_vals[*,1]-gaussian_y_vals[*,0]) / gaussian_dec_corr
 
-      ;Gaussian source model major and minor axes RA DEC
-      ;           _ a_2,d_2
-      ;          / \
-      ; a_3,d_3 |   | a_1,d_1
-      ;          \_/
-      ;   a_4,d_4
-      ;
-      ;Convert from deg to pixels
```

Comment:
Is this diagram no longer relevant? It would be helpful if you could update it rather than delete it if that is possible.

Reply:
The definition of the variables are not longer relevant because I removed the corresponding code. Bit unfortunate, really.